### PR TITLE
chore: removing `publishConfigs`, using `.d.ts` for types

### PR DESCRIPTION
**Pull request by arboleya**
# TL;DR

This PR favors `.d.ts` files (over `.ts` ones) as type definitions, even for local development. This requires rebuilding the workspace to see changes but increases reliability and compatibility with other tools.

 - [x] Upgrade `tsup`
 - [x] Refactor all `tsup` configs
 - [x] Add `postbuilld` script ([code](https://github.com/FuelLabs/fuels-ts/pull/1055/files#diff-21462feba05a79008c54d9a3267569b27b8897540ac86e47110e57f25402d224))
 - [x] Refactor all `package.json` configs
 - [x] Stops using `tsup` for generating types
 - [x] Start using `tsc` for generating types w/ Declaration Maps
 - [x] Create a new app in `internal` directory for validating configs

> **Note** This problem arose when I was working on #1049, which is blocked until this is merged.

# Summary

The attempt to use `.ts` files instead of `.d.ts` for development _(introduced by #984)_, didn't work as expected. VSCode only fully worked for **_main_** entry points _(not secondary ones)_, and other apps/bundlers would malfunction locally—_probably due to the format discrepancies between DTS and TS_.

Because of this, I'm bringing back the use of Declaration Maps files (`d.ts.map`), from my first experiments in #827. Although it adds an extra step, it is more reliable and compatible with other tools. (sorry @camsjams, I tried a lot)

## This approach offers
 1. Single build _(not more `publishConfig` overrides)_
    - _Local packages are 100% identical to published ones!_
 1. Increased reliability (local builds works like published ones)
 1. Increased compatibility with other tools _(NextJS, CRA, Vite)_

## BUT
 1. We need to build the repo to see the changes
    - Command `pnpm dev` automates this
    - TurboRepo cache helps a bit with speed
    - We can further reduce the over-fragmentation of packages in the future

# Final Notes

If you can, please review this other PR before this one:
 - #1056 

I separated that into another PR to make it easier to review this PR without the bulkiness of the demo apps, but I will need to merge that PR into this one before sending it to `master`.

### CODE REVIEW

**Review by Torres-ssf — DISMISSED** [review]

**Review by luizstacio — APPROVED** [review]
Lgtm

File: internal/check-imports/src/imports.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import * as abiCoder from '@fuel-ts/abi-coder';
+import * as abiTypegen from '@fuel-ts/abi-typegen';
+import * as address from '@fuel-ts/address';
+import * as asm from '@fuel-ts/asm';
+import * as contract from '@fuel-ts/contract';
+// forc-bin
+// fuels-gauge
+// fuels
```
Comment (Dhaiwat10) [inline internal/check-imports/src/imports.ts:8]: Did you forget to add these imports?

**Review by Dhaiwat10 — APPROVED** [review]

**Review by danielbate — DISMISSED** [review]

**Review by Torres-ssf — DISMISSED** [review]
:rocket:

**Review by LuizAsFight — DISMISSED** [review]

File: internal/check-imports/src/imports.ts (hunk 1 above)
Comment (arboleya) [reply internal/check-imports/src/imports.ts]: Good catch! 🕵️

Yes and no:
 - The `fuels-gauge` and `forc-bin` do not export anything
 - My assumption for leaving the `fuels` out was a mistake — added back [here](https://github.com/FuelLabs/fuels-ts/pull/1055/commits/af8ec90ab0fc88019d1c4af2a9af7bc6f8bb91cc)

Thank you 🙏

**Review by Torres-ssf — APPROVED** [review]

**Review by digorithm — APPROVED** [review]
LGTM
